tests/manage/z_cluster/test_scc.py

```diff
@@ -0,0 +1,226 @@
+
+        # create objects under performance directory
+        cmd = "cd performance && for i in $(seq 0 1000000);do dd if=/dev/urandom of=object_$i bs=512 count=1;done"
+        simple_app_pod.exec_sh_cmd_on_pod(command=cmd, timeout=5400)
+
+        # get node where the simple-app pod scheduled
+        node = get_pod_node(simple_app_pod).name
+        logger.info(f"{simple_app_pod.name} pod is scheduled on node {node}")
+
+        # get the node mount
+        node_mount = (
+            OCP()
+            .exec_oc_debug_cmd(node=node, cmd_list=[f"df -h | grep {pv.name}"])
+            .split()[5]
+        )
+        logger.info(f"Node Mount: {node_mount}")
+
+        total_time = 0
+        for i, mode in enumerate(permission_map.keys()):
+            logger.info(f"PERMISSION: {mode}")
+            cmd = f"chmod {mode} {node_mount}"
+            OCP().exec_oc_debug_cmd(node=node, cmd_list=[cmd])
+
+            cmd = f"ls -ld {node_mount}"
+            mode_before = OCP().exec_oc_debug_cmd(node=node, cmd_list=[cmd]).split()[0]
+            logger.info(f"Mode before: {mode_before}")
+
+            # scale down simple-app deployment to 0
+            simple_app_dc.scale(replicas=0)
+
+            assert (
+                simple_app_dc.replicas == 0
+            ), "Failed to scale down simple-app deployment"
+
+            while OCP(kind="pod").get(
+                resource_name=simple_app_pod.name, dont_raise=True
+            ):
+                logger.info(f"{simple_app_pod.name} is still terminating; Retrying")
+
+            # fetch the time before scale up
+            time_before = datetime.now()
+
+            # scale up simple-app deployment to 1
+            simple_app_dc.scale(replicas=1)
+            assert (
+                simple_app_dc.replicas == 1
+            ), "Failed to scale up simple-app deployment"
+            simple_app_pod = Pod(
+                **get_pods_having_label(
+                    label="app=simple-app", namespace="test-project"
+                )[0]
+            )
+            try:
+                helpers.wait_for_resource_state(
+                    resource=simple_app_pod,
+                    state=constants.STATUS_RUNNING,
+                    timeout=timeout,
+                )
+            except Exception:
+                logger.info(
+                    f"Pod {simple_app_pod.name} didn't reach Running state within expected time"
+                )
+                raise
+            logger.info("simple-app deployment is scaled up to replica 1")
+
+            # fetch the time after scale up and add the difference to permission_map
+            time_after = datetime.now()
+            permission_map[mode] = (time_after - time_before).total_seconds() / 60
+            total_time += permission_map[mode] * 60
+
+            # maximum allowed time for the pods to come up is 2 times the avg time taken with other permission
+            timeout = (total_time / (i + 1)) * 2
```

Comment:
what is this time calculation?

Reply:
here i'm trying to generate the maximum timeout for the pods to come up in the next iteration. Its basically two times the current average time taken for all previous iteration. I felt this is a better idea instead of hard coding the timeout, so we make sure there is no unrealistic delay in the pods reconcile.